### ISSUE TITLE
no-comments-in-text-node seems to be broken with @typescript-eslint/parser

### ISSUE THREAD

**Issue by Axnyff** [yannickcr/eslint-plugin-react#2599]
Hello,

I've recently had had a bug with a comment text node and realizer the rule does not seem to work while using @typescript-eslint/parser.

I've created a draft pr which seems to fix the problem while not breaking the existing behavior: 
https://github.com/Axnyff/eslint-plugin-react/commit/d47233c486cd0423a90a8d81884db22ad391a800

Disclaimer: I had this warning while testing
```
WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: ~3.1.1

YOUR TYPESCRIPT VERSION: 3.8.3
```
but unless I'm mistaken, it doesn't matter in my case.

Thanks for your help.

**Comment by ljharb** [comment]
A PR with a failing test would be most helpful.

**Comment by Axnyff** [comment]
Hi, I've created this PR: https://github.com/yannickcr/eslint-plugin-react/pull/2601.

It seems that this snippet would fix the bug https://github.com/yannickcr/eslint-plugin-react/commit/13673b5a71f98c6167b64b5727ea101bb51aec69
